atom/browser/api/atom_api_app.cc

```diff
@@ -275,6 +276,53 @@ void App::SelectClientCertificate(
         cert_request_info->client_certs[0].get());
 }
 
+bool App::CanCreateWindow(const GURL& opener_url,
+                          const GURL& opener_top_level_frame_url,
+                          const GURL& source_origin,
+                          WindowContainerType container_type,
+                          const std::string& frame_name,
+                          const GURL& target_url,
+                          const content::Referrer& referrer,
+                          WindowOpenDisposition disposition,
+                          const blink::WebWindowFeatures& features,
+                          bool user_gesture,
+                          bool opener_suppressed,
+                          content::ResourceContext* context,
+                          int render_process_id,
+                          int opener_render_view_id,
+                          int opener_render_frame_id,
+                          bool* no_javascript_access) {
+  DCHECK_CURRENTLY_ON(content::BrowserThread::IO);
+
+  content::BrowserThread::PostTask(content::BrowserThread::UI, FROM_HERE,
```

Comment:
We can put this logic in `AtomBrowserClient::CanCreateWindow`, so we only need to have `App::OnCreateWindow` .